### PR TITLE
Update zlib and freetype

### PR DESCRIPTION
**Pull request by Legimet**
Also, the zlib.a used by the zehn_loader is now built from source. It works fine with --gc-sections, so I added that (tested on an old Ndless without Zehn support, both with and without compression). The Freetype sample works as well.

### CODE REVIEW

File: ndless-sdk/tools/zehn_loader/Makefile [hunk 1]
```diff
@@ -1,18 +1,28 @@
 GXX := "$(shell (which arm-elf-g++ arm-none-eabi-g++ arm-linux-gnueabi-g++ | head -1) 2>/dev/null)"
 OBJCOPY := "$(shell (which arm-elf-objcopy arm-none-eabi-objcopy arm-linux-gnueabi-objcopy | head -1) 2>/dev/null)"
+CHOST := $(shell (echo $(shell basename $(GXX)) | sed 's/-g++$$//'))
 
-GXXFLAGS := -g -Os -Wall -Wextra -march=armv5te -fPIE -std=c++11 -fno-rtti -fno-exceptions -Wl,-Tldscript -nostdlib -nostartfiles -I ../../include -D _LDBL_EQ_DBL
+GXXFLAGS := -g -Os -Wall -Wextra -march=armv5te -fPIE -std=c++11 -fno-rtti -fno-exceptions -Wl,-Tldscript -Wl,--gc-sections -nostdlib -nostartfiles -I ../../include
+ZLIB_CFLAGS := -Os -fPIE -ffunction-sections -fdata-sections -DNO_GZIP -c
```
Comment (Vogtinator) [inline ndless-sdk/tools/zehn_loader/Makefile]: What is the ```-c``` needed for?

File: ndless-sdk/tools/zehn_loader/Makefile [hunk 2]
```diff
@@ -1,18 +1,28 @@
 GXX := "$(shell (which arm-elf-g++ arm-none-eabi-g++ arm-linux-gnueabi-g++ | head -1) 2>/dev/null)"
 OBJCOPY := "$(shell (which arm-elf-objcopy arm-none-eabi-objcopy arm-linux-gnueabi-objcopy | head -1) 2>/dev/null)"
+CHOST := $(shell (echo $(shell basename $(GXX)) | sed 's/-g++$$//'))
```
Comment (Vogtinator) [inline ndless-sdk/tools/zehn_loader/Makefile]: What is CHOST used for here? Normally CHOST is the prefix of the host toolchain, which is most likely not arm.

File: ndless-sdk/tools/zehn_loader/Makefile (hunk 1 above)
Comment (Legimet) [inline ndless-sdk/tools/zehn_loader/Makefile]: A workaround for the zlib configure script to work correctly before the rest of the SDK is built.

File: ndless-sdk/tools/zehn_loader/Makefile (hunk 2 above)
Comment (Legimet) [inline ndless-sdk/tools/zehn_loader/Makefile]: That's how you get the zlib configure script to cross-compile.

File: ndless-sdk/tools/zehn_loader/Makefile (hunk 1 above)
Comment (Legimet) [inline ndless-sdk/tools/zehn_loader/Makefile]: Actually, I think there is a better solution.

Comment (Legimet) [inline ndless-sdk/tools/zehn_loader/Makefile]: Never mind, this seems to be the easiest way to do it.

**Pull request merged**